packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -327,4 +343,50 @@ App.propTypes = {
     config: PropTypes.object
 }
 
+/**
+ * a hook that parallelly and individually fetches category based on the given ids
+ * @param ids - list of categories ids to fetch
+ * @param queryOptions -  react query options
+ * @return list of react query results
+ */
+export const useCategoryBulk = (ids, queryOptions) => {
+    if (!ids || ids.length === 0) {
+        return
+    }
+    const api = useCommerceApi()
+    const {getTokenWhenReady} = useAccessToken()
+    const {
+        app: {commerceAPI}
+    } = getConfig()
+    const {
+        parameters: {organizationId}
+    } = commerceAPI
+    const {site} = useMultiSite()
+
+    const queries = ids.map((id) => {
+        return {
+            queryKey: [
+                '/organizations/',
+                organizationId,
+                '/categories/',
+                id,
+                {id, levels: 2, organizationId, siteId: site.id}
+            ],
+            queryFn: async () => {
+                const token = await getTokenWhenReady()
+                const res = await api.shopperProducts.getCategory({
+                    parameters: {id, levels: 2},
+                    headers: {
+                        authorization: `Bearer ${token}`
+                    }
+                })
+                return res
+            },
```

Comment:
```suggestion
            // Also needed:
            // import {useAuthorizationHeader} from 'commerce-sdk-react-preview/hooks/useAuthorizationHeader'
            // const getCategory = useAuthorizationHeader(async (opts) => await api.shopperProducts.getCategory(opts))
            queryFn: async () => await getCategory({parameters: {id, levels: 2}}),
```
We have marked `useAuthorizationHeader` as `@internal`, but we don't necessarily need to (or we could just ignore the warning).

Reply:
If we mark it as internal, itis not meant to be consumed by the public, we should not use it outside of the context of commerce-react imo. If we want to use it outside, then we should make it public